qadence_protocols/measurements/protocols.py

```diff
@@ -1,115 +1,69 @@
-            "tomography": self.verify_tomography,
-        }
-
-    def verify_options(self) -> dict:
-        return self.OPTIONS_TO_VERIFY[self.protocol]()
-
-    def verify_classical_shadow(self) -> dict:
-        """Extract shadow_size, accuracy and confidence from options."""
-
-        shadow_size = self.options.get("shadow_size", None)
-        accuracy = self.options.get("accuracy", None)
-        if shadow_size is None and accuracy is None:
-            raise KeyError(
-                "Shadow protocol requires either an option"
-                " 'shadow_size' of type 'int' or 'accuracy' of type 'float'."
-            )
-        confidence = self.options.get("confidence", None)
-        if confidence is None:
-            raise KeyError("Shadow protocol requires an option 'confidence' of type 'float'.")
-
-        return {"shadow_size": shadow_size, "accuracy": accuracy, "confidence": confidence}
-
-    def verify_tomography(self) -> dict:
-        n_shots = self.options.get("n_shots")
-        if n_shots is None:
-            raise KeyError("Tomography protocol requires a 'n_shots' kwarg of type 'int').")
-
-        return {"n_shots": n_shots}
-
-    def verify_robust_shadow(self) -> dict:
-        """Extract shadow_size, accuracy and confidence from options."""
-
-        shadow_size = self.options.get("shadow_size", None)
-        if shadow_size is None:
-            raise KeyError("Robust Shadow protocol requires an option 'shadow_size' of type 'int'.")
-        shadow_groups = self.options.get("shadow_groups", None)
-        if shadow_groups is None:
-            raise KeyError("Shadow protocol requires an option 'shadow_groups' of type 'int'.")
-
-        calibration = self.options.get("calibration", None)
+class Measurements(Protocol):
+    """Define a measurement protocol using a `MeasurementManager`.
 
-        return {
-            "shadow_size": shadow_size,
-            "shadow_groups": shadow_groups,
-            "calibration": calibration,
-        }
+    Possible options are available via the `MeasurementProtocols` type.
+    The `MeasurementManager` instance has a verify_options
 
-
-@dataclass
-class Measurements(Protocol):
-    TOMOGRAPHY = "tomography"
-    SHADOW = "shadow"
-    ROBUST_SHADOW = "robust_shadow"
+    Attributes:
+        protocol (str): Protocol name.
+        options (dict): Options to run protocol.
+    """
 
     def __init__(self, protocol: str, options: dict = dict()) -> None:
-        verified_options = MeasurementOptions(protocol, options).verify_options()
+        try:
+            module = importlib.import_module(PROTOCOL_TO_MODULE[protocol][0])
+            proto_class = getattr(module, PROTOCOL_TO_MODULE[protocol][1])
+        except (KeyError, ModuleNotFoundError, ImportError) as e:
+            raise type(e)(f"Failed to import Mitigations due to {e}.")
+        self.manager: MeasurementManager = proto_class(data=None, options=options)
```

Comment:
I am a bit confused: isn't the role of the `MeasurementManager` to _produce_ `MeasurementData` through for instance the `measure` method ? If so, why passing `data` as `None` to the initialiser ?

Reply:
The role of the manager is not only to produce `MeasurementData` but also provide extra functionalities to postprocess it. In principle, one could obtain data from somewhere else, and feed it as an input, but here None means no data was obtained yet.

Reply:
Indeed, but in practice, there is no place where you can externally pass `data` ? `Measurements` doesn't provide it and I am not sure `MeasurementManager` is exposed to the user ?

Reply:
This is used for instance in testing faster that robust shadows are equal to the vanilla shadows when no calibration has been done.

Reply:
hm could you point me where ?

Reply:
No right, the current way was to do:
`robust_shadows.manager.data = shadow_measurements.manager.data`

I propably should set a property to access and a setter for data.

Reply:
🤔

Reply:
Using a setter now with more transparency in Measurements